workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Components/OptionTests.cs

```diff
@@ -147,15 +147,15 @@ public void Equality_operator_works()
         {
             var payload = true;
             var option = new Option<bool>(payload);
-            Assert.AreEqual(true, option == option);
+            Assert.AreEqual(true, option == payload);
```

Comment:
This actually tests for something different now I think

Reply:
Icky, it just looks like a mistake

Reply:
This test encompasses the old one. It is not possible for this to fail and the old one to pass.

Reply:
Also given the old test was called `Equality_operator_works` the old implementation looks incorrect.

Reply:
`Option<T> == T` vs `Option<T> == Option<T>` are two different operations and should be tested independently imo